presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
+                return "varchar(" + varcharType.getLength() + ")";
+            }
+            else if (varcharType.getLength() <= 65535) {
+                return "text";
```

Comment:
Yeah - that is problematic. I do not see a very nice solution to that. Let me try to craft something.

**Presto->Mysql mapping**

I would map Presto's `VARCHAR(x)` up to `MAX_IN_ROW_VARCHAR_LENGTH` (which should be overridable by session property) directly to MySQLs `VARCHAR(x)`.
For larger values of `x` I would calculate `L` upper bound on number of bytes used by the values (i.e. `L=x*4`). And then calculate MysqlType as following:
 * `L <= 2^8` -> `TINYTEXT`
 * `L <= 2^16` -> `TEXT`
 * `L <= 2^24` -> `MEDIUMTEXT`
 * `L <= 2^32` -> `LONGTEXT`

**Mysql -> Presto mapping**
 * `VARCHAR(x)` -> `VARCHAR(x)`
 * `TINYTEXT` -> `VARCHAR(2^8-1)`
 * `TINY` -> `VARCHAR(2^16-1)`
 * `MEDIUMTEXT` -> `VARCHAR(2^24-1)`
 * `LONGTEXT` -> `VARCHAR`

Optionally we can skip support for `LONGTEXT` as this exceeds precision currently supported by presto for now.

It is worth noting that above mapping make sense (at least some sense) as long as we do not have INSERT support in JDBC connector. With INSERT we would be faced with the fact that inserted value could not fit in MySQL table even if it seems fine from perspective of Presto semantic analyzer.

@martint What do you think?